package/kedro_viz/api/rest/responses.py

```diff
@@ -419,12 +419,12 @@ def save_api_main_response_to_fs(main_path: str, remote_fs: Any):
         raise exc
 
 
-def save_api_node_response_to_fs(nodes_path: str, remote_fs: Any):
+def save_api_node_response_to_fs(nodes_path: str, remote_fs: Any, preview: bool):
     """Saves API /nodes/{node} response to a directory."""
     for nodeId in data_access_manager.nodes.get_node_ids():
         try:
             write_api_response_to_fs(
-                f"{nodes_path}/{nodeId}", get_node_metadata_response(nodeId), remote_fs
+                f"{nodes_path}/{nodeId}", get_node_metadata_response(nodeId, preview), remote_fs
```

Comment:
we can try setting a class variable for DataNodeMetadata instead of passing is_preview to the class as this variable is common for all instances. In DataNodeMetadata have -

```python
_is_preview_enabled_for_all_nodes: ClassVar[bool] = False

    @classmethod
    def set_preview_enabled_for_all_nodes(cls, value: bool):
        cls._is_preview_enabled_for_all_nodes = value

    @classmethod
    def is_preview_enabled_for_all_nodes(cls):
        return cls._is_preview_enabled_for_all_nodes
 ```  
 Before the for loop begins (at 424), we can set the class variable by doing
 
 ```python
 DataNodeMetadata.set_preview_enabled_for_all_nodes(is_preview_enabled)
 ```

Reply:
@ravi-kumar-pilla I didn't use `is_preview_enabled` as  `cls._is_preview_enabled` can be accessible within the class, but all the other changes I added. Thanks.